packages/react/src/transactions/useFollow.ts

```diff
@@ -102,24 +103,23 @@ export function useFollow(): UseDeferredTask<
   const { data: session } = useSession();
   const followProfile = useFollowController();
 
-  // const hasPendingUnfollowTx = useHasPendingTransaction(
-  //   isUnfollowTransactionFor({ profileId: profile.id }),
-  // );
+  return useDeferredTask(({ profile }) => {
+    invariant(
+      profile.operations.canFollow === TriStateValue.Yes,
+      "You can't follow this profile. Check the `profile.operations.canFollow` beforehand.",
+    );
 
-  return useDeferredTask(async ({ profile }) => {
-    // if (hasPendingUnfollowTx) {
-    //   return failure(
-    //     new PrematureFollowError(
-    //       `A previous unfollow request for ${profile.id} is still pending. Make sure you check 'profile.operations.canFollow' beforehand.`,
+    // if (!profile.operations.isFollowedByMe.isFinalisedOnchain) {
+    //   return Promise.resolve(
+    //     failure(
+    //       new PrematureFollowError(
+    //         `A previous unfollow request for ${profile.id} is still pending.
+    //       Check 'profile.operations.isFollowedByMe.isFinalisedOnchain' beforehand.`,
+    //       ),
     //     ),
     //   );
     // }
```

Comment:
You think this is not needed because of the `profile.operations.canFollow` and the invariant check above?

Reply:
no, the type PrematureFollowError conflicts with other error types, I don't get why, jut TS is not happy